nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureDataLakeStorage.java

```diff
@@ -192,4 +211,29 @@ static void uploadContent(DataLakeFileClient fileClient, InputStream in, long le
 
         fileClient.flush(length);
     }
+
+    private DataLakeFileClient renameFile(String fileName, String directoryPath, DataLakeFileClient fileClient, boolean overwrite) {
+        try {
+            final DataLakeRequestConditions destinationCondition = new DataLakeRequestConditions();
+            if (!overwrite) {
+                destinationCondition.setIfNoneMatch("*");
+            }
+            String destinationPath = StringUtils.isNotEmpty(directoryPath)
+                    ? directoryPath + "/" + fileName
+                    : fileName;
+            return fileClient.renameWithResponse(null, destinationPath, null, destinationCondition, null, null).getValue();
+        } catch (DataLakeStorageException dataLakeStorageException) {
+            getLogger().error("Error while renaming temp file on Azure Data Lake Storage", dataLakeStorageException);
```

Comment:
What behavior is expected for the temp file if rename is unsuccessful?

Reply:
@MuazmaZ thanks for your help, temp file will remain on the server and will be overwritten at next try.